### PR TITLE
[SDK] Perf improvements for erc-20 methods

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Continue work for #1648 

## Changes made

- [ ] Public API changes: list the public API changes made if any
- [x] Internal API changes: explain the internal logic changes

## How to test

- [x] Automated tests: link to unit test file
- [ ] Manual tests: step by step instructions on how to test

### CODE REVIEW

File: packages/sdk/src/evm/core/classes/erc-20-history.ts [hunk 1]
```diff
@@ -55,14 +55,24 @@ export class TokenERC20History {
...
+    );
     return Promise.all(
       Object.keys(balances).map(async (addr) => ({
         holder: addr,
-        balance: await fetchCurrencyValue(
-          this.contractWrapper.getProvider(),
-          this.contractWrapper.address,
-          balances[addr],
-        ),
+        balance: balanceData[addr],
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/core/classes/erc-20-history.ts:75]: this `map` doesnt need to be async anymore right?

File: packages/sdk/src/evm/core/classes/erc-20-history.ts [hunk 2]
```diff
@@ -55,14 +55,24 @@ export class TokenERC20History {
         balances[to] = balances[to].add(amount);
       }
     });
+    const _promises = Object.entries(balances).map(async ([addr, value]) => {
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/core/classes/erc-20-history.ts:58]: do we have a unit test for this one?

File: packages/sdk/src/evm/core/classes/erc-20-history.ts (hunk 1 above)
Comment (kien-ngo) [reply packages/sdk/src/evm/core/classes/erc-20-history.ts:75]: Oops you're right

File: packages/sdk/src/evm/core/classes/erc-20-history.ts (hunk 2 above)
Comment (kien-ngo) [reply packages/sdk/src/evm/core/classes/erc-20-history.ts:58]: I think we do: https://github.com/thirdweb-dev/js/blob/main/packages/sdk/test/evm/token.test.ts#L65

File: packages/sdk/src/evm/core/classes/erc-20-history.ts (hunk 1 above)
Comment (kien-ngo) [reply packages/sdk/src/evm/core/classes/erc-20-history.ts:75]: This code looks off. I'm working on it

Comment (kien-ngo) [reply packages/sdk/src/evm/core/classes/erc-20-history.ts]: @joaquim-verges would you be able to take a look at the code again? Thank you

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**